prov/psm/src/psmx.h

```diff
@@ -451,7 +451,10 @@ struct psmx_trigger {
 };
 
 struct psmx_fid_cntr {
-	struct fid_cntr		cntr;
+	union {
+		struct fid_cntr		cntr;
+		struct util_cntr	util_cntr; /* for util_poll_run */
+	};
```

Comment:
Yes for both. 

This patch is the simplest way to get it work correctly. A refactoring is needed to get all the fields sorted out, including resolving the duplication of the util_domain structure pointers.